torch/csrc/jit/codegen/cuda/python_frontend/fusion_record.h

```diff
@@ -305,92 +882,315 @@ struct OutputRecord : RecordFunctor {
+      output = Nvf::IrBuilder::create<Nvf::ComplexDouble>();
+    } else if (dtype_ == Nvf::DataType::Bool) {
+      output = Nvf::IrBuilder::create<Nvf::Bool>();
+    } else if (dtype_ == Nvf::DataType::Int) {
+      output = Nvf::IrBuilder::create<Nvf::Int>();
     } else {
       TORCH_CHECK(false, "Dtype is not supported:", dtype_);
     }
     fd.addInput(output);
-    fd.setFusionState(outputs.at(0), output);
+    fd.setFusionState(outputs_.at(0).index, output);
+  }
+
+  virtual void print(std::ostream& os, bool close_function = true) const {
+    RecordFunctor::print(os, false);
+    os << "dtype=" << dtypeToPyString(dtype_);
+    if (close_function) {
+      os << ")";
+    }
   }
 
  private:
   //! Scalar data type.
-  NvfDataType dtype_;
+  Nvf::DataType dtype_;
 };
 
-//! Specialized Record Functor for the FusionDefinition's var op.
+//! Specialized Record Functor for recording FusionDefinition Start.
+//! There should only ever be one instance of this Record in the
+//! Fusion Cache.
 
-struct VarianceOpRecord : RecordFunctor {
-  VarianceOpRecord(
-      std::vector<size_t> _args,
-      std::vector<size_t> _outputs,
+struct StartRecord : RecordFunctor {
+  StartRecord() : RecordFunctor({}, {}, "start", RecordType::Start) {}
+  virtual ~StartRecord() = default;
+  virtual RecordFunctor* clone() final {
+    return new StartRecord(*this);
+  }
+
+  //! Child specific hash function in lower 32 bits.
+  //! | 31 ---------------------------------------  0 |
+  //! | None                                          |
+  virtual size_t hash() const final {
+    return RecordFunctor::hash();
+  }
+
+  virtual bool operator==(const RecordFunctor& other) const final {
+    auto result = false;
+    if (dynamic_cast<const StartRecord*>(&other)) {
+      result = RecordFunctor::operator==(other);
+    }
+    return result;
+  }
+
+  virtual void operator()(FusionDefinition& fd) final {}
+};
+
+//! Specialized Record Functors for Normalization based ops.
+
+struct NormOpRecord : RecordFunctor {
+  NormOpRecord(
+      std::vector<State> args,
+      std::vector<State> outputs,
+      std::string name,
+      RecordType type,
       std::vector<int>& axes,
       int64_t correction,
       bool keep_dim)
-      : RecordFunctor(std::move(_args), std::move(_outputs)),
+      : RecordFunctor(std::move(args), std::move(outputs), name, type),
         axes_(axes),
         correction_(correction),
         keep_dim_(keep_dim) {}
-  virtual ~VarianceOpRecord() = default;
+  virtual ~NormOpRecord() = default;
+  virtual RecordFunctor* clone() = 0;
 
-  void operator()(FusionDefinition& fd) final {
-    auto arg = fd.getFusionState(args.at(0))->as<NvfTensorView>();
-    auto output =
-        torch::jit::fuser::cuda::variance(arg, axes_, correction_, keep_dim_);
-    fd.setFusionState(outputs.at(0), output);
+  // I am skipping the bassel's correction value in the hash because
+  // I suspect we might change it to a bool from a 64-bit value
```

Comment:
It's required to be an integer and not just bool by the interface. It's not just 0 or 1 value but any positive integer really.